src/status_im/contexts/wallet/create_account/new_keypair/check_your_backup/view.cljs

```diff
@@ -62,26 +61,32 @@
 
 (defn- view-internal
   []
-  (let [random-indices  (random-selection)
-        quiz-index      (reagent/atom 0)
-        incorrect-count (reagent/atom 0)
-        show-error?     (reagent/atom false)]
+  (let [random-indices                        (random-selection)
+        quiz-index                            (reagent/atom 0)
+        incorrect-count                       (reagent/atom 0)
+        show-error?                           (reagent/atom false)
+        {:keys [secret-phrase random-phrase]} (rf/sub [:wallet/create-account])]
     (fn []
-      (let [current-word-index        (get random-indices (min @quiz-index (dec questions-count)))
-            current-word              (get temp/secret-phrase current-word-index)
-            [options-r-0 options-r-1] (random-words-with-string temp/random-words current-word)
-            on-button-press           (fn [word]
-                                        (if (= word current-word)
-                                          (do
-                                            (reset! quiz-index (inc @quiz-index))
-                                            (reset! incorrect-count 0)
-                                            (reset! show-error? false))
-                                          (do
-                                            (when (> @incorrect-count 0)
-                                              (rf/dispatch [:show-bottom-sheet
-                                                            {:content cheat-warning}]))
-                                            (reset! incorrect-count (inc @incorrect-count))
-                                            (reset! show-error? true))))]
+      (let [current-word-index            (get random-indices
+                                               (min @quiz-index (dec questions-count)))
+            current-word                  (get secret-phrase current-word-index)
+            [options-row-0 options-row-1] (random-words-with-string random-phrase current-word)
+            on-button-press               (fn [word]
+                                            (if (= word current-word)
+                                              (do
+                                                (when (< @quiz-index 4)
```

Comment:
what is `4` related to here?

Reply:
@J-Son89 number of questions. Refactored it to the defined constant